### ISSUE TITLE
Duplicate username preventing startup with no info

### ISSUE THREAD

**Issue by Arcanemagus** [Tautulli/Tautulli#276]
I'm not entirely sure _how_, but I ended up with a user in the database with no `user_id`, attempting to start PlexPy gave the following traceback. (For anyone running into this I simply deleted the user entry to get PlexPy running again.)

The issue is that the log provides no information on _which_ user account is giving the error, I put the following line in plextv.py to debug the issue:

``` python
logger.info("uid: %s, un: %s" % (item['user_id'], item['username']))
```

But a more elegant solution would catch the following error properly (I'm not sure why it isn't already), and log the offending account.

```
08-Nov-2015 16:46:24 - ERROR :: MainThread : Fatal Error executing UPDATE users SET username = ?, custom_avatar_url = ?, thumb = ?, is_allow_sync = ?, is_restricted = ?, is_home_user = ?, email = ? WHERE user_id = ? :: UNIQUE constraint failed: users.username
08-Nov-2015 16:46:24 - ERROR :: MainThread : Uncaught exception: Traceback (most recent call last):
  File "/usr/local/share/plexpy/PlexPy.py", line 232, in <module>
    main()
  File "/usr/local/share/plexpy/PlexPy.py", line 157, in main
    plexpy.initialize(config_file)
  File "/usr/local/share/plexpy/plexpy/__init__.py", line 175, in initialize
    plextv.refresh_users()
  File "/usr/local/share/plexpy/plexpy/plextv.py", line 57, in refresh_users
    monitor_db.upsert('users', new_value_dict, control_value_dict)
  File "/usr/local/share/plexpy/plexpy/database.py", line 124, in upsert
    self.action(update_query, value_dict.values() + key_dict.values())
  File "/usr/local/share/plexpy/plexpy/database.py", line 77, in action
    sql_result = c.execute(query, args)
IntegrityError: UNIQUE constraint failed: users.username

Traceback (most recent call last):
  File "/usr/local/share/plexpy/PlexPy.py", line 232, in <module>
    main()
  File "/usr/local/share/plexpy/PlexPy.py", line 157, in main
    plexpy.initialize(config_file)
  File "/usr/local/share/plexpy/plexpy/__init__.py", line 175, in initialize
    plextv.refresh_users()
  File "/usr/local/share/plexpy/plexpy/plextv.py", line 57, in refresh_users
    monitor_db.upsert('users', new_value_dict, control_value_dict)
  File "/usr/local/share/plexpy/plexpy/database.py", line 124, in upsert
    self.action(update_query, value_dict.values() + key_dict.values())
  File "/usr/local/share/plexpy/plexpy/database.py", line 77, in action
    sql_result = c.execute(query, args)
sqlite3.IntegrityError: UNIQUE constraint failed: users.username
```

**Comment by drzoidberg33** [comment]
Thanks for the report.

**Comment by JonnyWong16** [comment]
I believe I fixed this in v1.3.0 by removing the unique constraint on the username (only a unique user_id is required now).

Reopen the issue if it shows up again.
